src/main/java/io/supertokens/webserver/api/passwordless/ConsumeCodeAPI.java

```diff
@@ -113,6 +113,8 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         } catch (StorageTransactionLogicException | StorageQueryException | NoSuchAlgorithmException
                 | InvalidKeyException e) {
             throw new ServletException(e);
+        } catch (IllegalArgumentException ex) {
+            throw new ServletException(new BadRequestException("Input encoding error"));
```

Comment:
We should not catch `IllegalArgumentException` and convert that to `BadRequestException` cause `IllegalArgumentException` can be thrown from MANY places and we don't want all of them to map to a bad request.

So we should isolate the places where we expect base64 encoding to fail, throw a custom compile time exception there and map that to bad request